ecs-agent/wsclient/client_test.go

```diff
@@ -422,3 +446,37 @@ func TestCtxCancel(t *testing.T) {
 	cancel()
 	assert.EqualError(t, <-messageError, "context canceled")
 }
+
+func TestPeriodicDisconnect(t *testing.T) {
+	closeWS := make(chan []byte)
+	defer close(closeWS)
+
+	mockServer, _, requests, errChan, _ := utils.GetMockServer(closeWS)
+	mockServer.StartTLS()
+
+	types := []interface{}{ecsacs.AckRequest{}}
+	cs := getTestClientServer(mockServer.URL, types, 1)
+	// Setting up a lower disconnect timer value for testing.
+	timer, err := cs.Connect(mockDisconnectTimeoutMetricName, 10*time.Second, 2*time.Second)
+	require.NoError(t, err)
+	defer timer.Stop()
+	assert.True(t, cs.IsReady(), "expected websocket connection to be ready")
+
```

Comment:
The test from this line in below seems to be somewhat of a hack and potentially overcomplicated. Could we not just wait for the max possible amount of time it would take for the client to reach its periodic disconnect logic, and then assert that the websocket connection was in fact closed by checking against what is returned by calling `cs.IsConnected()`?

Reply:
Yeah what I meant by "hack" was that the current way the test is designed is using some other indirect means to test by proxy the functionality (i.e., making a bunch of requests that will take longer to complete than the max connection time and waiting until we receive an error) when we could likely just test the functionality directly and with less unnecessary overhead (i.e., making sure the websocket connection is disconnected after the max connection time).

In this case I meant for "hack" to be more or less synonymous with overcomplicated -  apologies if that was unclear!